applications/OptimizationApplication/custom_utilities/geometrical/symmetry_utility.cpp

```diff
@@ -0,0 +1,272 @@
+        }
+    }
+
+    KRATOS_INFO("SymmetryUtility::Initialize: ") << " Finished initialization of symmetry utility "<<mUtilName<<" in " << timer.ElapsedSeconds() << " s." << std::endl;    
+}
+
+SymmetryUtility::NodeTypePointer SymmetryUtility::GetRotatedNode(NodeType& rNode, int rotation_index){
+    NodeTypePointer p_new_node = Kratos::make_intrusive<NodeType>(rNode.Id(), rNode[0], rNode[1], rNode[2]);
+
+    const array_3d v1 = rNode.Coordinates()-mRotationalSymmetryData.Point;
+    p_new_node->Coordinates() = prod(mRotationalSymmetryData.RotationMatrices[rotation_index], v1) + mRotationalSymmetryData.Point;
+
+    return p_new_node;
+}
+
+SymmetryUtility::NodeTypePointer SymmetryUtility::GetReflectedNode(NodeType& rNode){
+    NodeTypePointer p_new_node = Kratos::make_intrusive<NodeType>(rNode.Id(), rNode[0], rNode[1], rNode[2]);
+    const array_3d v = rNode.Coordinates() - mPlaneSymmetryData.Point;
+    p_new_node->Coordinates() = prod(mPlaneSymmetryData.ReflectionMatrix, v) + mPlaneSymmetryData.Point;
+    return p_new_node;
+}
+
+void SymmetryUtility::Update()
+{
+
+}
+
+void SymmetryUtility::GetRotationMatrix(double angle, Matrix& rRotMat){
+    rRotMat.resize(3,3);
+    rRotMat = ZeroMatrix(3,3);
+
+    const double c=cos(angle*Globals::Pi/180);
+    const double s=sin(angle*Globals::Pi/180);
+    const double t = 1-c;
+    const array_3d& k = mRotationalSymmetryData.Axis;
+
+    rRotMat(0,0) = t*k[0]*k[0] + c;         rRotMat(0,1) = t*k[0]*k[1] - k[2]*s;    rRotMat(0,2) = t*k[0]*k[2] + k[1]*s;
+    rRotMat(1,0) = t*k[0]*k[1] + k[2]*s;    rRotMat(1,1) = t*k[1]*k[1] + c;         rRotMat(1,2) = t*k[1]*k[2] - k[0]*s;
+    rRotMat(2,0) = t*k[0]*k[2] - k[1]*s;    rRotMat(2,1) = t*k[1]*k[2] + k[0]*s;    rRotMat(2,2) = t*k[2]*k[2] + c;
+}
+
+void SymmetryUtility::ApplyOnVectorField( const Variable<array_3d> &rNodalVariable )
+{
+    if(mAxisSymmetry){        
+        std::vector<Vector> aux;
+        #pragma omp parallel
+        {
+            std::vector<Vector> aux_private;
+            #pragma omp for nowait schedule(static)
+            for(auto& r_pair : mRotationalSymmetryData.Map) {
+                Vector vec_val = r_pair.first->FastGetSolutionStepValue(rNodalVariable);
+                Matrix rot_mat;
+                for(int n_c = 0; n_c<r_pair.second.size();n_c++){
+                    GetRotationMatrix((n_c+1)*mRotationalSymmetryData.Angle,rot_mat);
+                    vec_val += prod(trans(rot_mat),r_pair.second[n_c]->FastGetSolutionStepValue(rNodalVariable));
+                }                
+                    
+                vec_val /= (r_pair.second.size()+1);
+                aux_private.push_back(vec_val);
+            }
+            #pragma omp for schedule(static) ordered
+            for(int i=0; i<omp_get_num_threads(); i++) {
+                #pragma omp ordered
+                aux.insert(aux.end(), aux_private.begin(), aux_private.end());
+            }
+        }
```

Comment:
This seems like a reduction to a list? Am I wrong? If it, then i would use `ParallelUtilities` with `ReductionUtilities`
Is there a specific reason why you use `nowait` and `static schedule` in here?

The reasoning is, we want to switch to c++ based parallelism at one point, and this will not allow it.